### PR TITLE
ModuleObjcMessageTrace: also include the source file name where an ObjcMethod is referenced

### PR DESCRIPTION
**Pull request by nkcsgexi**
rdar://138776708

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: test/IDE/objc_send_collector.swift [hunk 1]
```diff
@@ -15,4 +15,5 @@ public func testProperties(_ x: FooClassBase) {
 // CHECK-DAG: fooBaseInstanceFunc0
 // CHECK-DAG: fooBaseInstanceFunc1
 // CHECK-DAG: "type": "FooClassBase"
-// CHECK-DAG: SOURCE_DIR/test/IDE/Inputs/mock-sdk/Foo.framework/Headers/Foo.h
+// CHECK-DAG: "declared_at": "SOURCE_DIR/test/IDE/Inputs/mock-sdk/Foo.framework/Headers/Foo.h
+// CHECK-DAG: "referenced_at": "SOURCE_DIR/test/IDE/objc_send_collector.swift"
```
Comment (bnbarham) [inline test/IDE/objc_send_collector.swift:19]: There's going to be *heaps* of duplication here 😅

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: test/IDE/objc_send_collector.swift (hunk 1 above)
Comment (nkcsgexi) [reply test/IDE/objc_send_collector.swift]: Consolidated all source file paths to a side table.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci Please clean test Linux platform

**Comment by nkcsgexi** [comment]
@swift-ci Please clean test Windows platform

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci Please clean test Windows platform

**Comment by nkcsgexi** [comment]
@swift-ci Please clean test Windows platform

**Pull request merged**